doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,136 @@
+       - Wait for signal to stabilize on PHY   
+    c) Enable optical module (turn laser on/ enable tx) : xcvrd or platform bootstrap/infra
+
+  In boards not having PHY, #b) not needed but #a) and #c) sequence to be followed.
+  
+  ## Clause from QSFP-DD (CMIS4.0 spec)
+  
+     Excerpt from CMIS4.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![61f5b485-cf3b-4ca8-beac-9102b6feabfe](https://user-images.githubusercontent.com/69485234/147173702-f124fc9d-ef27-4816-b1a1-b4a44a5833a7.PNG)
+
+
+  ## Clause from QSFP-DD (CMIS5.0 spec)
+  
+     Excerpt from CMIS5.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![96a35dc5-618f-418c-9593-5639a90f1b28](https://user-images.githubusercontent.com/69485234/147173164-5ad0123c-479a-4774-b3ee-12a81fdd7d7e.PNG)
+     
+
+# Objective
+
+Have a determistic approach for Interface link bring-up sequence i.e. below sequence to be followed:
+  1. Initialize and enable NPU Tx and Rx path
+  2. For system with 'External' PHY: Initialize and enable PHY Tx and Rx on both line and host sides; ensure host side link is up 
+  3. Then only perform optics data path initialization/activation/Tx enable (for 400G) and Tx enable (for 100G) 
+
+# Proposal
+
+Recommend following this high-level work-flow sequence to accomplish the Objective:
+- xcvrd to subscribe to a new field “host_tx_ready” in port table state-DB
+- Orchagent will set the “host_tx_ready” to true/false based on the SET_ADMIN_STATE attribute return status to syncd/gbsyncd. (As part of SET_ADMIN_STATE attribute enable, the NPU Tx is enabled)
+- xcvrd process the “host_tx_ready” value change event and do optics datapath init / de-init using CMIS API
+- Recommendation is to follow this proposal for all the known interface speeds - 400G/100G/40G/25G/10G
+  - 400G - as mentioned above the CMIS spec to be followed
+  - 100G/40G/25G/10G - deterministic approach to bring the interface will eliminate any link stability issue which will be difficult to chase in the production network
+  - This synchronization will also benefit native 10G SFPs interfaces as they are "plug N play" and may not have quiescent functionality. (xcvrd can use the optional 'soft tx disable' ctrl reg to disable the tx)
+
+# Proposed Work-Flows
+Please refer to the  flow/sequence diagrams which covers the following required use-cases
+  - Transceiver initialization
+  - admin enable configurations 
+  - admin disable configurations
+
+# Transceiver Initialization 
+  (at platform bootstrap layer)
+  
+  ![LC boot-up sequence - optics INIT (platform bootstrap)](https://user-images.githubusercontent.com/69485234/147166795-5665670d-dd2b-4b6f-976c-eabcc65d5448.png)
+
+# Applying 'interface admin startup' configuration
+
+![LC boot-up sequence - 'admin enable' Config gets applied](https://user-images.githubusercontent.com/69485234/147166867-56f3e82d-1b1c-4b7a-a867-5470ee6050e7.png)
```

Comment:
Isn't set_port_attribute only blocking in sync mode?

Reply:
right

Reply:
@mikeberesford 
Yes, Sync mode is a pre-requisite for this HLD. 

With Async mode, the expectation is the set ADMIN_STATE attribute update in ASIC-DB (Syncd/GBsyncd) return status will be treated to set the host_tx_ready in Orchagent.
We will update the HLD with this information.

As per the understanding from MSFT-Sonic team and Sonic chassis work group discussion meeting, SONiC-NOS operates in Sync mode and continue to remain same.

Reply:
HLD updated